proposals/permissions-requestSiteAccess-api.md

```diff
@@ -0,0 +1,160 @@
+# permissions.[add/remove]SiteAccessRequest() API
+
+## API Overview
+
+### Background
+
+Extensions can request host permissions via:
+- Manifest declared `host_permissions`, granted at install time by default
+- Manifest declared `optional_host_permissions`, granted at runtime after a user gesture by `<browser>.permissions.request()`
+- Manifest declared `matches` for a `content_script`, granted at install time by default
+
+In some browsers, users can withhold host permissions causing the extension to only have site access when the extension is invoked. For example, in Chrome a user can set an extension to run only "on click". When the extension is clicked, it gains site access to the tab's main frame origin (effectively acting like [`activeTab` permission](https://developer.chrome.com/docs/extensions/develop/concepts/activeTab#what-activeTab-allows).
+
+Each browser may decide how it signals the user when an extension is requesting access (e.g through the extensions menu). However, there is no way for an extension to explicitly signal at runtime it’s requesting site access after it was withheld without a user gesture and a heavy-weight permission dialog.
+
+
+### Objective
+
+Allow the extension to show site access requests at runtime without any user gesture in a less-obtrusive way than with `permissions.request()`. This can be done with a new API that:
+
+- Applies to a specific tab or document id. Developers can specify inactive tabs (request will be shown by the browser when the user activated them).
+- Doesn’t need to be made inside the handler for a user action
+- Shows the request in the UI, handled differently by each browser. See more in [UI Elements and User-Visible Effects](#ui-elements-and-user-visible\-effects) section
+- When accepted, grants always access to the site’s top origin
+- Resets the request on cross-origin navigation
+
+#### Use Cases
+
+An extension requested access to a site but the user withheld its access and forgot about it. Extension wants to signal it needs site access to execute without user action. For example, “shopping” extension wants to show site access when user navigates to “amazon.com” and access was withheld.
+
+### Consumers
+
+Extensions that want to signal the user when they need site access.
+
+## Implementation
+
+### API Schema
+
+```
+// Adds a site access request. Request will only be signaled to the user if
+// extension can be granted access to site in the request (i.e., one specified
+// in host_permissions, optional_host_permissions, a content script match
+// pattern, or an applicable activeTab site).
+// Resolves whether the request is valid.
+Promise<{bool}> <browser>.permissions.addSiteAccessRequest(
```

Comment:
The `{bool}` here looks incorrect. I think @Rob--W was intending it to be around the arguments, not the promise result.